### DISCUSSION TITLE
How to check which toolchain is used?

### DISCUSSION THREAD

**Discussion by qudix** [xmake-io/xmake#2950] [Q&A]
Is there a way to check if a target is using a specific toolchain? 

For example, only add certain flags if not using clang-cl:
```lua
on_config(function(target)
    ....
    if target:is_plat("windows") then
        target:add("cxxflags", "/permissive-")
        if not target_using_clang_cl then
            target:add("cxxflags", "/Zc:externConstexpr")
        end
    end
    ....
end)
```

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake/blob/a6eb410b9880f43349f8da4f1787d68fc5fb08af/xmake/rules/c%2B%2B/modules/modules_support/common.lua#L108

**Comment by qudix** [comment]
This seems to work:
```lua
if not a_target:has_tool("cxx", "clang_cl") then
```
